File: backend/src/config/init_db.py
import sqlite3
import os
import sys

def main():
    db_path = "ascendiq.db"
    schema_path = "backend/src/config/schema.sql"

    if not os.path.exists(schema_path):
        print(f"Error: Schema file not found at {schema_path}")
        sys.exit(1)

    print(f"Connecting/Creating SQLite Database: {db_path}...")
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()

    print(f"Reading schema file: {schema_path}...")
    with open(schema_path, 'r') as f:
        schema_sql = f.read()

    print("Executing schema...")
    try:
        cursor.executescript(schema_sql)
        conn.commit()
        print("Database Schema initialized successfully inside SQLite!")
    except Exception as e:
        print(f"Error executing database schema: {e}")
        conn.close()
        sys.exit(1)

    cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
    tables = cursor.fetchall()
    print("\nCreated Tables:")
    for t in tables:
        print(f" - {t[0]}")

    conn.close()

if __name__ == "__main__":
    main()

File: backend/src/config/init_db.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import db, { pool } from './db.js';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function initDb() {
  const schemaPath = path.resolve(__dirname, './schema.sql');
  console.log(`Reading SQL schema from ${schemaPath}...`);
  const schemaSql = fs.readFileSync(schemaPath, 'utf8');

  try {
    console.log('Initializing schema inside PostgreSQL/Supabase...');
    await pool.query(schemaSql);
    console.log('PostgreSQL schema initialized successfully!');
  } catch (err) {
    console.error('Error executing PostgreSQL database schema:', err);
  } finally {
    await db.close();
  }
}

initDb();
export default initDb;
